src/cube.js

```diff
@@ -344,6 +344,83 @@ class Cube {
         throw new Error(`getSingleData: no such measure ${measureId}`);
     }
 
+    /* 
+    * This function returns an array of all possible combinations of dimension items
+    * It takes an array of dimension ids to exclude from the combinations generation process
+    */
+    scan(excludeDimensionIds, rootAttribute, cb) {
+        const combinations = getCombinations(
+            this.getDimensionItemsMap(excludeDimensionIds),
+        );
+
+        combinations.forEach((combination) => {
+            const dicedCube = this.diceByDimensionsItem(combination, rootAttribute);
+            cb(dicedCube, combination);
+        });
+    }
+
+    /*
+    * This function takes an array of dimension ids and returns a new cube with the specified dimensions sliced by the specified dimension items
+    */
+    aggregateByDimensions(dimensions) {
+        return this.dimensionIds
+            .filter((d) => !dimensions.includes(d))
+            .reduce((acc, dimension) => {
+                return acc.slice(dimension, 'all', 'all');
+            }, this);
+    }
+
+    /*
+    * This function returns an object with dimension id as key and dimension items as value
+    * It takes an array of dimension ids to exclude from the result
+    */
+    getDimensionItemsMap(excludeDimensionIds) {
+        const dimensions = excludeDimensionIds?.length
+            ? this.dimensionIds.filter((d) => excludeDimensionIds.includes(d))
+            : this.dimensionIds;
+        return dimensions.reduce(
+            (acc, cur) => ({
+                ...acc,
+                [cur]: this.getDimension(cur).getItems(),
+            }),
+            {},
+        )
+    }
+
+    /*
+    * This function takes dimensions and rootAttribute as arguments and returns a new cube with diced dimensions
+    */
+    diceByDimensionsItem(dimensions, rootAttribute) {
+        return Object.keys(dimensions).reduce((acc, dimension) => {
+            const values = [dimensions[dimension]].flat();
+            if (values.length > 0 && acc.dimensionIds.includes(dimension)) {
+                return acc.dice(dimension, rootAttribute, values);
+            }
+
+            return acc;
+        }, this);
+    }
```

Comment:
so you don't have to pass in the `rootAttribute` as a parameter here necessarily (and it wouldn't be correct either, as it might be that the dimensions which you provided thosed share the same rootAttribute), you can access it through the original dimension object and.
This is what you could do instead:
```suggestion
    diceByDimensionItems(dimensionMap) {
        return Object.keys(dimensionMap).reduce((acc, dimension) => {
            const dimensionItems = [dimensionMap[dimension]].flat();
            if (dimensionItems.length > 0 && acc.dimensionIds.includes(dimension)) {
                const rootAttribute = this.getDimension(dimension).rootAttribute;
                return acc.dice(dimension, rootAttribute, dimensionItems);
            }

            return acc;
        }, this);
    }
```